src/main/webapp/scripts/controller-auth.js

```diff
@@ -0,0 +1,65 @@
+import { postVideoComment } from '/scripts/post-comment.js';
+import { videoId, postCommentElement, failedInitCallback, successfulApiCallback, failedApiCallback, updateSigninStatus } from '/scripts/view.js';
+
+/** API key provied by GCP to authenticate the client */
+const apiKey = '';
+/** client ID provied by GCP to identify the client */
+const clientId = '';
+/** discovery docs used to identify API endpoints */
+const discoveryDocs = ['https://www.googleapis.com/discovery/v1/apis/youtube/v3/rest'];
+/** OAuth2 scope required to post a comment */
+const scope = 'https://www.googleapis.com/auth/youtube.force-ssl';
+
+/** Load the client from Google API oauth2.0 JS library */
+function handleClientLoad() {
+  gapi.load('client:auth2', initClient);
+}
+
+/** 
+ * Initialize the client with appropiate fields.
+ * And listen for changed in the signin status.
+ */
+function initClient() {
+  return gapi.client.init({
+    apiKey,
+    discoveryDocs,
+    clientId,
+    scope
+  })
+    .then(function () {
+      // Listen for sign-in state changes.
+      gapi.auth2.getAuthInstance().isSignedIn.listen((isSignedIn) =>
+        updateSigninStatus(isSignedIn, handleAuthClick, handleSignoutClick, handleCommentClick));
+
+      // Handle the initial sign-in state.
+      updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get(),
+        handleAuthClick, handleSignoutClick, handleCommentClick);
+    })
+    .catch(failedInitCallback);
+}
+
+/** 
+ * Handler that initiates the authorization flow with a popup.
+ * The user only needs to authorize the site once.
+ */
+function handleAuthClick(event) {
+  gapi.auth2.getAuthInstance().signIn();
+}
+
+/** Handler that quietly signs the user out, but does not revoke the authorization */
+function handleSignoutClick(event) {
+  gapi.auth2.getAuthInstance().signOut();
+}
+
+/** Handler that posts the comment and processes the result */
+function handleCommentClick(event) {
+  return postVideoComment(videoId, postCommentElement.value)
+    .then(successfulApiCallback)
+    .catch(failedApiCallback);
+}
+
+// Entry point
+handleClientLoad();
+
+// For testing purpose
+export { initClient, handleAuthClick, handleSignoutClick, handleCommentClick };
```

Comment:
Is there a way to test this module without exposing everything in this file? When you export in a file you are making an interface telling others that they can use these when that might be unintended.

Reply:
handleClientLoad / initClick are already somehow exposed because they are run onload. 

And I agree about encapsulating {handleAuthClick, handleSignoutClick, handleCommentClick} -- actually I have tried the natural way of invoking handleCommentClick like this

```
postCommentButton.click()

assert.dom()...
```

but `postCommentButton.click()` isn't an aync function so `async/await` doesn't work as intended in the tests (setTimeOut works most of the time but flaky).

Reply:
Sorry I meant if there is a way to only expose the "public" methods only. 

For the click issue, idk if what I'm about to say is a hack. How about, using a queryselector to get the button, then somehow retrieve the function/callback that is called when the button is clicked, then override the click function to return a promise that calls the function/callback you retrieved, and then you can do .click() and await? lol idk if that is how this works haha.

Reply:
Yap it worked! The only thing is that `await button.click()` doesn't work as expected so I changed it to `await button.onclick()` wasn't so sure about the difference ...